model/infinispan/src/main/java/org/keycloak/jgroups/protocol/KEYCLOAK_JDBC_PING2.java

```diff
@@ -34,6 +42,107 @@ public class KEYCLOAK_JDBC_PING2 extends JDBC_PING2 {
 
     private JpaConnectionProviderFactory factory;
 
+    @Override
+    protected void handleView(View new_view, View old_view, boolean coord_changed) {
+        // If we are the coordinator, it is good to learn about new entries that have been added before we delete them.
+        // If we are not the coordinator, it is good to learn the new entries added by the coordinator.
+        // This avoids a "JGRP000032: %s: no physical address for %s, dropping message" that leads to split clusters at concurrent startup.
+        learnExistingAddresses();
+
+        // This is an updated logic where we do not call removeAll but instead remove those obsolete entries.
+        // This avoids the short moment where the table is empty and a new node might not see any other node.
+        if (is_coord) {
+            if (remove_old_coords_on_view_change) {
+                Address old_coord = old_view != null ? old_view.getCreator() : null;
+                if (old_coord != null)
+                    remove(cluster_name, old_coord);
+            }
+            Address[] left = View.diff(old_view, new_view)[1];
+            if (coord_changed || update_store_on_view_change || left.length > 0) {
+                writeAll(left);
+                if (remove_all_data_on_view_change) {
+                    removeAllNotInCurrentView();
```

Comment:
I think this is fine for our use-case, but would we need a new flag for this to make it into JGroups as we're no longer removing all data on a view change?